UnitTests/AppAuthTV/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,279 @@
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  XCTAssertEqualObjects(
+      ((OIDTVServiceConfiguration *)authRequest.configuration).TVAuthorizationEndpoint,
+      [self testServiceConfiguration].TVAuthorizationEndpoint);
+
+  OIDAuthorizationRequest *authRequestCopy = [authRequest copy];
+
+  XCTAssertEqualObjects(
+      ((OIDTVServiceConfiguration *)authRequestCopy.configuration).TVAuthorizationEndpoint,
+      [self testServiceConfiguration].TVAuthorizationEndpoint);
+}
+
+/*! @brief Tests the @c NSSecureCoding implementation by round-tripping an instance through the
+ * coding process and checking to make sure the source and destination both contain the
+ * @c TVAuthorizationEndpoint
+ */
+- (void)testSecureCoding {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  XCTAssertEqualObjects(
+      ((OIDTVServiceConfiguration *)authRequest.configuration).TVAuthorizationEndpoint,
+      [self testServiceConfiguration].TVAuthorizationEndpoint);
+
+  NSData *data = [NSKeyedArchiver archivedDataWithRootObject:authRequest];
+  OIDAuthorizationRequest *authRequestCopy = [NSKeyedUnarchiver unarchiveObjectWithData:data];
+
+  XCTAssertEqualObjects(
+      ((OIDTVServiceConfiguration *)authRequestCopy.configuration).TVAuthorizationEndpoint,
+      [self testServiceConfiguration].TVAuthorizationEndpoint);
+}
+
+/*! @brief Tests the @c URLRequest method on a request with no scopes or additional parameters
+ */
+- (void)testURLRequestBasicClientAuth {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  NSURLRequest *URLRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects(URLRequest.HTTPMethod, kHTTPPost);
+  XCTAssertEqualObjects([URLRequest valueForHTTPHeaderField:kHTTPContentTypeHeaderKey],
+                        kHTTPContentTypeHeaderValue);
+  XCTAssertEqualObjects(URLRequest.URL.absoluteString, kInitializerTestTVAuthEndpoint);
```

Comment:
In previous tests above you were using [self testServiceConfiguration].TVAuthorizationEndpoint to get the TVAuthEndpoint. Please make sure to use one of the two mechanisms and keep it consistent throughout the file.

Reply:
Good point, I decided to go with kTestTVAuthEndpoint everywhere for succinctness

Reply:
Although I feel like it would be maybe slightly better to go through the testServiceConfig since it in effect was testing that initializer

Reply:
Nevermind, decided to compare against testServiceConfiguration everywhere